scylla-cdc/src/stream_reader.rs

```diff
@@ -139,45 +139,7 @@ impl StreamReader {
             values[begin_index] = CqlValue::Timestamp(window_begin);
             values[end_index] = CqlValue::Timestamp(window_end);
 
-            let mut sleep_after_timeout = BASIC_TIMEOUT_SLEEP;
-
-            let mut next_state = None;
-            'try_query: loop {
-                let state_clone = next_state.clone();
-                let query_res = self
-                    .session
-                    .execute_paged(&query_base, &values, next_state)
-                    .await;
-                match query_res {
-                    Ok(x) => {
-                        sleep_after_timeout = BASIC_TIMEOUT_SLEEP;
-                        if let Some(rows) = x.rows {
-                            let schema = CDCRowSchema::new(&x.col_specs);
-
-                            for row in rows {
-                                consumer
-                                    .consume_cdc(CDCRow::from_row(row, &schema))
-                                    .await?;
-                            }
-                        }
-                        match x.paging_state {
-                            Some(state) => next_state = Some(state),
-                            None => break 'try_query,
-                        }
-                    },
-                    Err(QueryError::TimeoutError) => {
-                        sleep(time::Duration::from_millis(sleep_after_timeout)).await;
-                        sleep_after_timeout = sleep_after_timeout * TIMEOUT_FACTOR;
-                        if sleep_after_timeout >= TIMEOUT_LIMIT {
-                            println!("WARN: The database keeps returning a timeout error. Trying again...");
-                            sleep_after_timeout = TIMEOUT_LIMIT;
-                        }
-
-                        next_state = state_clone;
-                    },
-                    Err(e) => panic!("The database returned the following error: {:?}", e),
-                }
-            }
+            self.read_log(&query_base, &values, &mut consumer).await?;
```

Comment:
I'd recommend moving the logic out first, then add timeouts. This will make applying review suggestions easier (you won't have to fix the refactoring commit after modifying the other commit).